### PR TITLE
[MGPG-102] Drop MAT used in sign-and-deploy-file

### PR DESCRIPTION
**Pull request by cstamas**
Drop MAT, use Resolver, as we have min 3.2.5

---

https://issues.apache.org/jira/browse/MGPG-102

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/gpg/SignAndDeployFileMojo.java [hunk 1]
```diff
@@ -257,6 +219,15 @@ public class SignAndDeployFileMojo extends AbstractGpgMojo {
     @Parameter(property = "files")
     private String files;
 
+    /**
+     * @since 3.0.0
+     */
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
```
Comment (hboutemy) [inline src/main/java/org/apache/maven/plugins/gpg/SignAndDeployFileMojo.java:225]: why not @Component?

Comment (cstamas) [reply src/main/java/org/apache/maven/plugins/gpg/SignAndDeployFileMojo.java]: Coz did not want to mix in that change, so I did "as it was for the rest". anyway, pushed now a change that moves ALL to component.

**Review by hboutemy — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #245

**Comment by jira-importer** [comment]
Resolve #245

**Pull request merged**